src/main/kotlin/com/sourcegraph/cody/config/CodyAuthenticationManager.kt

```diff
@@ -80,12 +80,12 @@ class CodyAuthenticationManager(val project: Project) : Disposable {
    * to update their state accordingly later.
    */
   @CalledInAny
-  private fun getActiveAccountTier(forceRefresh: Boolean): CompletableFuture<AccountTier> {
+  private fun getActiveAccountTier(forceRefresh: Boolean): CompletableFuture<AccountTier?> {
     activeAccountTier?.let { if (!forceRefresh) return it }
 
     val account = getActiveAccount() ?: return CompletableFuture.completedFuture(null)
     val previousAccountTier = activeAccountTier?.getNow(null)
-    val accountTierFuture = CompletableFuture<AccountTier>()
+    val accountTierFuture = CompletableFuture<AccountTier?>()
```

Comment:
I do not understand how this can be completed with  `null`.
Where and in what way that completions happen?

Reply:
Line 86 is the only place where null can be returned. If the execution passes beyond that line, the result will be non-null, but the type of the future still has to match.

Reply:
Ah, it happens there:

`val account = getActiveAccount() ?: return CompletableFuture.completedFuture(null)`

So we had wrong return type definition type all along.